[Mobile Payments ]Refactor IPP Onboarding checks

toupper opened the pull request:
… statuses that can collect payments

<!-- Remember about a good descriptive title. -->

<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
With this PR we refactor the onboarding checks for In-Person Payments to allow payments only if the WCPay status is complete or restricted soon (with pending requirements skipped). This is different than the default last approach, where we only checked if the status was unknown to stop the flow.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
With a store where WCPay status is restricted soon, check that you can take payments. Check also if the status is complete.

@joshheald I invited you to my store where status is restricted soon, feel free to use it if necessary.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->


---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

toupper committed:
Be more defensive when allowing card present payments by checking all statuses that can collect payments

joshheald commented on WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingUseCase.swift:
```diff
@@ -474,8 +474,9 @@ private extension CardPresentPaymentsOnboardingUseCase {
 
-    func isInUndefinedState(account: PaymentGatewayAccount) -> Bool {
-        account.wcpayStatus == .unknown
+    func canCollectCardPresentPayments(account: PaymentGatewayAccount) -> Bool {
```
```suggestion
    func accountStatusAllowedForCardPresentPayments(account: PaymentGatewayAccount) -> Bool {
```

toupper replied:
As mentioned below, changed in 15b918d9f9e786d44943060f23e22b7ed5ff4658

joshheald commented on WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingUseCase.swift:
```diff
@@ -337,7 +337,7 @@ private extension CardPresentPaymentsOnboardingUseCase {
             }
         }
-        guard !isInUndefinedState(account: account) else {
+        guard canCollectCardPresentPayments(account: account) else {
```
```suggestion
        guard accountStatusAllowedForCardPresentPayments(account: account) else {
```
A bit of a nit-pick but it seems more specific to what we're checking

toupper replied:
Indeed, that's more specific, thanks for the comment, changed in 15b918d9f9e786d44943060f23e22b7ed5ff4658

joshheald reviewed (approved):
Works as expected – I suggested a rename but feel free to take or leave it.

joshheald committed:
Use passed currency on receipts

UK In-Person Payments receipts were showing with $ signs.

This was due to the currency code used from the API being `gbp` instead of `GBP`, which meant it didn’t match the `CurrencyCode.rawValue` when checked.

All our `CurrencyCode` values are upper case, so it seems reasonable to convert anything that’s passed in to upper case as well, to avoid this issue in future.

toupper committed:
Merge pull request #9489 from woocommerce/task/use-passed-in-currency-for-receipts

[Mobile Payments] Use passed currency on receipts

toupper committed:
More exact name
